keen/api.py

```diff
@@ -126,12 +127,70 @@ def post_events(self, events):
         self._error_handling(response)
         return self._get_response_json(response)
 
+
+    def _order_by_is_valid_or_none(self, params):
+        """
+        Validates that a given order_by has proper syntax.
+
+        :param params: Query params.
+        :return: Returns True if either no order_by is present, or if the order_by is well-formed.
+        """
+        if not "order_by" in params or not params["order_by"]:
+            return True
+
+        def _order_by_dict_is_not_well_formed(d):
+            if not isinstance(d, dict):
+                # Bad type.
+                return True
+            if "property_name" in d and d["property_name"]:
+                if "direction" in d and not (d["direction"] == "ASC" or d["direction"] == "DESC"):
+                    # Bad direction provided.
+                    return True
+                for k in d:
+                    if k != "property_name" and k != "direction":
+                        # Unexpected key.
+                        return True
+                # Everything looks good!
+                return False
+            # Missing required key.
+            return True
+
+        # order_by is converted to a list before this point if it wasn't one before.
+        order_by_list = json.loads(params["order_by"])
```

Comment:
It's unfortunate that by the time we validate this, we've already serialized to JSON only to immediately deserialize back *from* JSON just to sanitize this input :/  It seems like a waste of cycles, though I realize to avoid that, this check would have to be higher up the call chain before `get_params()` is called and so would have to be called in several places instead of just in `query()`.  Just something to think about.

Reply:
Yeah, I don't think the serialization performance cost is going to be ugly enough to merit repeating this check in several places.

Reply:
Fair enough. It might be worth leaving a comment though...something saying that it was an intentional choice to do so.